File: 1109.corpFlightBookings.go
func corpFlightBookings(bookings [][]int, n int) []int {
    arr := make([]int , n+2)
    for _, book := range bookings {
        arr[book[0]] += book[2]
        arr[book[1]+1] -= book[2]
    }
    for i := 1; i <= n; i++ {
        arr[i] += arr[i-1]
    }
    return arr[1:n+1]
}

File: 1109.corpFlightBookings.cpp
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> diff(n+2, 0);
        vector<int> ans;

        for (auto b: bookings) {
            diff[b[0]] += b[2];
            diff[b[1]+1] -= b[2];
        }

        int seats = 0;
        for (int i = 1; i <= n; i++) {
            seats += diff[i];
            ans.push_back(seats);
        }

        return ans;
    }
};
